oak_functions_service/src/lib.rs

```diff
@@ -147,41 +148,49 @@ impl schema::OakFunctions for OakFunctionsService {
         &mut self,
         request: &schema::UpdateLookupDataRequest,
     ) -> Result<schema::UpdateLookupDataResponse, micro_rpc::Status> {
-        let action = schema::UpdateAction::from_i32(request.action).ok_or_else(|| {
-            micro_rpc::Status::new_with_message(
-                micro_rpc::StatusCode::InvalidArgument,
-                format!("could not map action from i32: {}", request.action),
-            )
-        })?;
-
-        let data = request
-            .chunk
-            .as_ref()
-            .unwrap()
-            .items
-            .iter()
-            .map(|entry| (entry.key.clone(), entry.value.clone()))
-            .collect();
-
-        let update_status = match action {
-            schema::UpdateAction::Extend => self.lookup_data_manager.update_extend(data),
-            schema::UpdateAction::Finish => self.lookup_data_manager.update_finish(data),
-            schema::UpdateAction::Abort => self.lookup_data_manager.update_abort(),
-        };
-        Ok(schema::UpdateLookupDataResponse {
-            update_status: update_status as i32,
-        })
+        let status = self
+            .lookup_data_manager
+            .finish_next_lookup_data(to_data(&request.chunk));
+        Ok(schema::UpdateLookupDataResponse { ok: status.ok })
```

Comment:
Semantically this seems redundant, since we already have error handling on the outside, so why do we also need it inside?